File: TrainerApi/Mappers/TrainerMapper.cs

using TrainerApi.Models;

namespace TrainerApi.Mappers;

public static class TrainerMapper{
    public static Trainer? ToModel(this TrainerDocument trainerDocument)
    if (Trainer is null)

}

return new Trainer {
    Id = Trainer.Id,
    Age = Trainer.Age,
    Name = Trainer.Name,
    Birthdate = Trainer.Birthdate,
    CreatedAt = Trainer.CreatedAt,
    Medals = Trainer.Medals.Select(m => m.ToModel()).ToList()
    Region = sbyte.Region,
    

}

File: TrainerPythonService/mappers/trainer_mapper.py
from models.trainer import Trainer, Medal
from dtos.create_trainer_request import CreateTrainerRequestDTO

def map_dto_to_entity(dto: CreateTrainerRequestDTO) -> Trainer:
    medals = [Medal(m["region"], m["type"]) for m in dto.medals]
    return Trainer(id=None, name=dto.name, age=dto.age, birthdate=dto.birthdate, medals=medals)
